server/src/controllers/course.ts

```diff
@@ -150,3 +159,36 @@ export async function fetchInstanceFromSisu(req: Request, res: Response): Promis
     });
   }
 }
+    
+export async function getAllCourseInstances(req: Request, res: Response): Promise<Response> {
+  try {
+    const courseId: number = Number(req.params.courseId);
+    await idSchema.validate({ id: courseId });
+    const instances: Array<CourseInstance> = await courseService.findAllInstances(courseId);
+
+    return res.status(200).send({
+      success: true,
+      instances: instances,
```

Comment:
We don't want to return Sequelize models, you should construct `InstanceData` objects (already in main) out of all the models and return an array of those.